source/retryable-reads/tests/unified/gridfs-download-serverErrors.yml

```diff
@@ -0,0 +1,421 @@
+description: gridfs-download-serverErrors
+
+schemaVersion: '1.9'
+
+runOnRequirements:
+  -
+    minServerVersion: '4.0'
+    topologies:
+      - single
+      - replicaset
+  -
+    minServerVersion: 4.1.7
+    topologies:
+      - sharded
+      - load-balanced
+
+createEntities:
+  -
+    client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      observeEvents:
+        - commandStartedEvent
+  -
+    database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database_name retryable-reads-tests
+  -
+    bucket:
+      id: &bucket0 bucket0
+      database: *database0
+
+initialData:
+  -
+    collectionName: &files_collection_name fs.files
+    databaseName: *database_name
+    documents:
+      -
+        _id:
+          $oid: '000000000000000000000001'
+        length: 1
+        chunkSize: 4
+        uploadDate:
+          $date: '1970-01-01T00:00:00.000Z'
+        filename: abc
+        metadata: {  }
+  -
+    collectionName: &chunks_collection_name fs.chunks
+    databaseName: *database_name
+    documents:
+      -
+        _id:
+          $oid: '000000000000000000000002'
+        files_id:
+          $oid: '000000000000000000000001'
+        'n': 0
+        data:
+          $binary:
+            base64: EQ== # hex: 11
+            subType: '00'
+
+tests:
+  -
+    description: 'Download succeeds after InterruptedAtShutdown'
+    operations:
+      -
+        object: testRunner
+        name: failPoint
+        arguments:
+          client: *client0
+          failPoint: &failCommand_failPoint
+            configureFailPoint: failCommand
+            mode:
+              times: 1
+            data:
+              failCommands:
+                - find
+              errorCode: 11600
+      - &retryable_operation
+        object: *bucket0
+        name: download
+        arguments:
+          id:
+            $oid: '000000000000000000000001'
+        expectResult:
+          $$matchesHexBytes: "11"
+    expectEvents:
+      -
+        client: *client0
+        events:
+          - &retryable_command_started_event
+            commandStartedEvent:
+              command:
+                find: *files_collection_name
+                filter:
+                  _id:
+                    $oid: '000000000000000000000001'
+              databaseName: *database_name
+          - *retryable_command_started_event
+          - &find_chunks_command_started_event
+            commandStartedEvent:
+              command:
+                find: *chunks_collection_name
+                # Avoid checking additional fields since the exact query may
+                # differ among drivers. expectResult is sufficient to assert
+                # correctness.
```

Comment:
Quoting the commit:

> Drivers may differ in how they query the chunks collection. For instance, [PHPLIB queries](https://github.com/mongodb/mongo-php-library/blob/1.17.0/src/GridFS/CollectionWrapper.php#L115) the chunk's n field with $gte in addition to files_id. Rather than use $$matchAsRoot (schema version 1.13+) to emulate the legacy test behavior of ignoring extra fields, the commandStartedEvent assertion was relaxed and expectResult was introduced to assert that the file is read successfully.

I know various drivers are behind on schema versions, so sticking with an older version seemed preferable. The chunk query also isn't vital to these tests, and `expectResult` with `$$matchesHexBytes` is overall an improvement IMO.